### PR TITLE
Minify Extensions

### LINKED ISSUE #497: Minify Extensions
With the change to extensions kev lost the ability to output only a few properties to the overlay file. 

Right now this is not possible because reconcile will only look at the values of composego.ServiceConfig.Extensions

https://github.com/appvia/kev/blob/5a1c05c568427024fb3ef76a9fc0a5d6bdd1e341/pkg/kev/manifest.go#L208

https://github.com/appvia/kev/blob/5a1c05c568427024fb3ef76a9fc0a5d6bdd1e341/pkg/kev/override.go#L188

Changing this behaviour could have some unexpected behaviour like the Inferred value from the composego.ServiceConfig in docker-compose.yaml being different from the previously rendered one in a way that makes the manifest non upgradeable. I am not sure if this is something that will happen in practice but I think it's noting here. 

The goal for this ticket is to find a way to remove the values that are the same when comparing the result of 

K8SCfgFromCompose and ParseK8SCfgFromMap since those could be "inflated" in a following command. 

For this I had thought of changing the code of mergo.Merge to essentially remove the *dst.Property when dst.Property== src.Property, this combined with omitempty tags for whatever properties are not required to show in the overlay should achieve the goal.

### PR DESCRIPTION
**Pull request by ezodude**
This resolves #497 

- Minifies environment overrides on init - including overwrite values received from the sources.

- Applies a just in time configuration strategy by overlaying config info at the last minute to render the required k8s manifests.

Config info is used from:
- Sources compose config.
- Any sources' k8s extensions.
- Minified environment overrides.

### CODE REVIEW

File: pkg/kev/config/minify.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		probeConfig = ProbeConfig{
+			HTTP: srcCfg.Workload.LivenessProbe.ProbeConfig.HTTP,
+		}
+	case ProbeTypeTCP.String():
+		probeConfig = ProbeConfig{
+			TCP: srcCfg.Workload.LivenessProbe.ProbeConfig.TCP,
+		}
+	}
+
+	return SvcK8sConfig{
```
Comment (marcinc) [inline pkg/kev/config/minify.go:42]: will JIT config respect any additional parameters user may include in environment override?

File: pkg/kev/manifest.go [hunk 2]
```diff
@@ -181,39 +188,83 @@ func (m *Manifest) ReconcileConfig(envs ...string) (*Manifest, error) {
...
+func validateEnvExtensions(e *Environment, base *composeOverride) error {
+	for _, s := range e.GetServices() {
+		baseSvc, missingSvcErr := base.getService(s.Name)
+		if missingSvcErr != nil {
+			continue
+		}
+
+		baseSvcK8sCfg, err := config.ParseSvcK8sConfigFromMap(baseSvc.Extensions, config.SkipValidation())
+		if err != nil {
+			return errors.Wrapf(missingSvcErr, "when parsing service %s extensions in base override", baseSvc.Name)
```
Comment (marcinc) [inline pkg/kev/manifest.go:227]: is it base or override? Am not sure I understand this message correctly.

File: pkg/kev/manifest.go [hunk 3]
```diff
@@ -181,39 +188,83 @@ func (m *Manifest) ReconcileConfig(envs ...string) (*Manifest, error) {
...
+
+	for name, vol := range e.GetVolumes() {
+		baseVol, missingVolError := base.getVolume(name)
+		if missingVolError != nil {
+			continue
+		}
+
+		baseVolK8sCfg, err := config.ParseVolK8sConfigFromMap(baseVol.Extensions)
+		if err != nil {
+			return errors.Wrapf(missingVolError, "when parsing vol %s extensions in base override", name)
```
Comment (marcinc) [inline pkg/kev/manifest.go:253]: similar comment here regarding "base override". I'm guessing we are talking about volume k8s config (via extension) in base compose file?

**Review by marcinc — COMMENTED** [review]
a few small comments

File: pkg/kev/config/minify.go (hunk 1 above)
Comment (ezodude) [reply pkg/kev/config/minify.go:42]: Yes, they get merged during render.

File: pkg/kev/manifest.go (hunk 3 above)
Comment (ezodude) [reply pkg/kev/manifest.go:253]: Yes that's exactly right: volume k8s config (via extension) in base compose file.

File: pkg/kev/manifest.go (hunk 2 above)
Comment (ezodude) [reply pkg/kev/manifest.go:227]: Same as volumes below. It's the service k8s config (via extension) in the base compose file.

File: pkg/kev/manifest.go (hunk 3 above)
Comment (marcinc) [reply pkg/kev/manifest.go:253]: That's what I figured out. Feels like "override" at the end is superfluous?

**Review by marcinc — APPROVED** [review]
👍   I left one tiny comment there but overall LGTM.

**Review by marcinc — APPROVED** [review]
👍

**Pull request merged**